x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/inference_input_form/index_input.tsx

```diff
@@ -0,0 +1,106 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useMemo } from 'react';
+
+import useObservable from 'react-use/lib/useObservable';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiSpacer,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiLoadingSpinner,
+  EuiText,
+} from '@elastic/eui';
+
+import { ErrorMessage } from '../../inference_error';
+import { extractErrorMessage } from '../../../../../../../common';
+import type { InferrerType } from '..';
+import { useIndexInput, InferenceInputFormIndexControls } from '../index_input';
+import { RUNNING_STATE } from '../inference_base';
+
+interface Props {
+  inferrer: InferrerType;
+}
+
+export const IndexInput: FC<Props> = ({ inferrer }) => {
+  const data = useIndexInput({ inferrer });
+  const { reloadExamples, selectedField } = data;
+
+  const [errorText, setErrorText] = useState<string | null>(null);
+  const runningState = useObservable(inferrer.runningState$);
+  const examples = useObservable(inferrer.inputText$) ?? [];
+  const outputComponent = useMemo(() => inferrer.getOutputComponent(), [inferrer]);
+  const infoComponent = useMemo(() => inferrer.getInfoComponent(), [inferrer]);
+
+  async function run() {
+    setErrorText(null);
+    try {
+      await inferrer.infer();
+    } catch (e) {
+      setErrorText(extractErrorMessage(e));
+    }
+  }
+
+  return (
+    <>
+      <>{infoComponent}</>
+      <InferenceInputFormIndexControls inferrer={inferrer} data={data} />
+
+      <EuiSpacer size="m" />
+
+      <EuiFlexGroup>
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            onClick={run}
+            disabled={runningState === RUNNING_STATE.RUNNING || selectedField === undefined}
+            fullWidth={false}
+          >
+            <FormattedMessage
+              id="xpack.ml.trainedModels.testModelsFlyout.inferenceInputForm.runButton"
+              defaultMessage="Test"
+            />
+          </EuiButton>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          {runningState === RUNNING_STATE.RUNNING ? <EuiLoadingSpinner size="xl" /> : null}
+        </EuiFlexItem>
+        <EuiFlexItem />
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            onClick={reloadExamples}
+            disabled={runningState === RUNNING_STATE.RUNNING || selectedField === undefined}
+          >
+            <FormattedMessage
+              id="xpack.ml.trainedModels.testModelsFlyout.inferenceInputForm.reloadButton"
+              defaultMessage="Reload examples"
+            />
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiSpacer size="m" />
+
+      {errorText !== null || runningState === RUNNING_STATE.FINISHED_WITH_ERRORS ? (
```

Comment:
Nit - probably worth adding a spacer after the error callout as well as above:

<img width="451" alt="image" src="https://user-images.githubusercontent.com/7405507/201657224-59a15f21-0840-4012-ad3c-d2b3444b0a08.png">